packages/react/src/components/SlashMenu/DefaultSlashMenu.tsx

```diff
@@ -14,69 +14,106 @@ export function DefaultSlashMenu<BSchema extends BlockSchema>(
   const [orderedItems, setOrderedItems] = useState<
     ReactSlashMenuItem<BSchema>[] | undefined
   >(undefined);
-  const [renderedItems, setRenderedItems] = useState<JSX.Element[] | undefined>(
-    undefined
-  );
+  const [loader, setLoader] = useState<JSX.Element | null>(null);
   const [selectedIndex, setSelectedIndex] = useState<number>(0);
-  const notFoundCount = useRef(0);
 
-  // Sets the items to render.
+  const prevQuery = useRef<string | undefined>(undefined);
```

Comment:
I think the code is getting a bit complicated now. Maybe the easiest way to do this is to ignore the result when the current query doesn't match the query of the incoming result?